### PR TITLE
✨ Add list functions for Mentions and Private Messages

### PR DESCRIPTION
**Pull request by i-be-snek**
Hello, 

I'm wondering if it's possible to contribute here by adding some more functionality.

### CODE REVIEW

**Comment by db0** [comment]
@NicKoehler can you review this as I'm busy atm?

**Comment by NicKoehler** [comment]
yes

File: pythorhead/mention.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            unread_only (bool).
+            sort? (SortType)
+            page? (int).
+            limit? (int)
+
+        Returns:
+            dict? mentions response
+        """
+
+        params = {"auth": self._requestor.get_auth_token()}
```
Comment (NicKoehler) [inline pythorhead/mention.py:30]: auth is automatically set by requestor, you don't need to do that here

File: pythorhead/private_message.py [hunk 2]
```diff
@@ -25,3 +25,25 @@ def create(
...
+        Args:
+            unread_only (bool).
+            page (int).
+            limit (int).
+            
+        Returns:
+            dict? private message response
+        """
+        json: dict[str, Any] = {key: value for key, value in locals().items() if value is not None and key != "self"}
+        params = {"auth": self._requestor.get_auth_token()}
```
Comment (NicKoehler) [inline pythorhead/private_message.py:47]: same as above

**Review by NicKoehler — CHANGES_REQUESTED** [review]

File: pythorhead/mention.py (hunk 1 above)
Comment (i-be-snek) [reply pythorhead/mention.py]: Okay, I figured out the issue. I should not have used `json` and `params` for a GET request, `params` is enough. I've fixed this and removed the extra methods I created to help fetch the token.

File: pythorhead/mention.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        Args: 
+            unread_only (bool).
+            sort? (SortType)
+            page? (int).
+            limit? (int)
+
+        Returns:
+            dict? mentions response
+        """
+        unread_only = 'true' if unread_only else 'false'
```
Comment (NicKoehler) [inline pythorhead/mention.py:29]: Is this really needed? python automatically parse booleans afaik

File: pythorhead/private_message.py [hunk 4]
```diff
@@ -25,3 +25,25 @@ def create(
...
+        Args:
+            unread_only (bool).
+            page (int).
+            limit (int) with a max of 50, defaults to 20.
+            
+        Returns:
+            dict? private message response
+        """
+        limit = 50 if limit > 50 else limit
+        unread_only = 'true' if unread_only else 'false'
```
Comment (NicKoehler) [inline pythorhead/private_message.py:47]: same as above

**Review by NicKoehler — APPROVED** [review]

File: pythorhead/mention.py (hunk 3 above)
Comment (i-be-snek) [reply pythorhead/mention.py:29]: Yes, when passed into params, the API will complain about it not being a string `true`/`false`. You can try passing it as a bool into the api and see the error it throws.

File: pythorhead/private_message.py (hunk 4 above)
Comment (i-be-snek) [reply pythorhead/private_message.py:47]: Wrote a comment above

File: pythorhead/mention.py (hunk 3 above)
Comment (i-be-snek) [reply pythorhead/mention.py:29]: Error:

```
Error encountered while Request.GET: Query deserialize error: provided string was not `true` or `false`
```

**Pull request merged**